public/app/features/alerting/unified/components/alert-groups/AlertDetails.tsx

```diff
@@ -13,37 +16,52 @@ interface AmNotificationsAlertDetailsProps {
 
 export const AlertDetails: FC<AmNotificationsAlertDetailsProps> = ({ alert, alertManagerSourceName }) => {
   const styles = useStyles2(getStyles);
+  const isExternalAM = !isGrafanaRulesSource(alertManagerSourceName);
   return (
     <>
       <div className={styles.actionsRow}>
-        {alert.status.state === AlertState.Suppressed && (
-          <LinkButton
-            href={`${makeAMLink(
-              '/alerting/silences',
-              alertManagerSourceName
-            )}&silenceIds=${alert.status.silencedBy.join(',')}`}
-            className={styles.button}
-            icon={'bell'}
-            size={'sm'}
-          >
-            Manage silences
-          </LinkButton>
-        )}
-        {alert.status.state === AlertState.Active && (
-          <LinkButton
-            href={makeLabelBasedSilenceLink(alertManagerSourceName, alert.labels)}
-            className={styles.button}
-            icon={'bell-slash'}
-            size={'sm'}
-          >
-            Silence
-          </LinkButton>
-        )}
-        {alert.generatorURL && (
-          <LinkButton className={styles.button} href={alert.generatorURL} icon={'chart-line'} size={'sm'}>
-            See source
-          </LinkButton>
-        )}
+        <Authorize
+          actions={
+            isExternalAM
+              ? [AccessControlAction.AlertingInstancesExternalWrite]
+              : [AccessControlAction.AlertingInstanceCreate, AccessControlAction.AlertingInstanceUpdate]
+          }
+        >
+          {alert.status.state === AlertState.Suppressed && (
+            <LinkButton
+              href={`${makeAMLink(
+                '/alerting/silences',
+                alertManagerSourceName
+              )}&silenceIds=${alert.status.silencedBy.join(',')}`}
+              className={styles.button}
+              icon={'bell'}
+              size={'sm'}
+            >
+              Manage silences
+            </LinkButton>
+          )}
+          {alert.status.state === AlertState.Active && (
+            <LinkButton
+              href={makeLabelBasedSilenceLink(alertManagerSourceName, alert.labels)}
+              className={styles.button}
+              icon={'bell-slash'}
+              size={'sm'}
+            >
+              Silence
+            </LinkButton>
+          )}
+        </Authorize>
+        <Authorize
+          actions={
+            isExternalAM ? [AccessControlAction.DataSourcesExplore] : [AccessControlAction.AlertingInstanceUpdate]
```

Comment:
`See source` does not update anything, does it? If no, then I think we should require read permissions. WDYT?